docs/design/assembly-loading.md

```diff
@@ -0,0 +1,83 @@
+# ILLink Assembly Loading
+
+We would like to support loading and processing assemblies later in the pipeline, so that dynamically referenced assemblies can be correctly processed. A secondary goal is to avoid unnecessary processing of unused assemblies.
```

Comment:
It'd be nice to improve the wording here. "We would like to" I think it's more than that, we need solution for specific problems. "A secondary goal is to avoid unnecessary processing of unused assemblies" needs reasons, etc.

Reply:
Add more details here